server/services/reviewService.test.ts

```diff
@@ -98,24 +111,62 @@ describe('reviewService', () => {
   })
 
   describe('getReports', () => {
+    const reportSummary = (id: number): ReportSummary => ({
+      id,
+      bookingId: id + 1,
+      reporterName: `reporter-${id}`,
+      offenderNo: `offender-${id}`,
+      incidentDate: new Date(id),
+    })
+
+    const incidentSummary = (id: number): IncidentSummary => ({
+      id,
+      bookingId: id + 1,
+      incidentdate: new Date(id),
+      staffMemberName: `reporter-${id}`,
+      isOverdue: false,
+      offenderName: `Prisoner prisoner-${id}`,
+      offenderNo: `offender-${id}`,
+    })
+
     test('it should call query on db', async () => {
-      const reportSummary = (id: number): ReportSummary => ({
-        id,
-        bookingId: id + 1,
-        reporterName: `reporter-${id}`,
-        offenderNo: `offender-${id}`,
-        incidentDate: new Date(id),
-      })
       const query = { prisonNumber: 'AA' }
 
       incidentClient.getIncompleteReportsForReviewer.mockResolvedValue([reportSummary(1)])
       incidentClient.getCompletedReportsForReviewer.mockResolvedValue([reportSummary(2)])
+      offenderService.getOffenderNames.mockResolvedValue(
+        Promise.resolve({
+          'offender-1': 'Prisoner prisoner-1',
+          'offender-2': 'Prisoner prisoner-2',
+        })
+      )
+
+      const result = await service.getReports('userName', 'agency-1', query)
+      expect(result).toEqual({ awaitingReports: [incidentSummary(1)], completedReports: [incidentSummary(2)] })
+
+      expect(incidentClient.getIncompleteReportsForReviewer).toBeCalledWith('agency-1', query)
+      expect(incidentClient.getCompletedReportsForReviewer).toBeCalledWith('agency-1', query)
+      expect(offenderService.getOffenderNames).toBeCalledWith('userName-system-token', ['offender-1', 'offender-2'])
+    })
+
+    test('it should filter on prisoner name', async () => {
+      const query = { prisonerName: 'OnER-2' }
+
+      incidentClient.getIncompleteReportsForReviewer.mockResolvedValue([reportSummary(1)])
+      incidentClient.getCompletedReportsForReviewer.mockResolvedValue([reportSummary(2)])
+      offenderService.getOffenderNames.mockResolvedValue(
+        Promise.resolve({
```

Comment:
If you do `mockResolvedValue` , you don't need to return a promise - it wraps it for you.

Reply:
Oh yes!  Bit obvious really. (from the name). I'll fix it...